test/component/physics/navierstokes/test_navierstokes.py

```diff
@@ -0,0 +1,155 @@
+import numpy as np
+import pytest
+import sys
+sys.path.append('../src')
+
+import physics.navierstokes.navierstokes as navierstokes
+import physics.navierstokes.tools as ns_tools
+import meshing.meshbase as meshbase
+
+rtol = 1e-15
+atol = 1e-15
+
+
+def test_diffusion_flux_2D():
+	'''
+	This tests the diffusive flux for a 2D case.
+	'''
+	mesh = meshbase.Mesh(ndims=2)
+	physics = navierstokes.NavierStokes2D(mesh)
+	physics.set_physical_params()
+	physics.get_transport = ns_tools.set_transport("Sutherland")
+
+	ns = physics.NUM_STATE_VARS
+
+	P = 101325.
+	rho = 1.1
+	u = 2.5
+	v = 3.5
+	gamma = physics.gamma
+	R = physics.R
+	rhoE = P / (gamma - 1.) + 0.5 * rho * (u * u + v * v)
+
+	Uq = np.zeros([1, 1, ns])
+
+	irho, irhou, irhov, irhoE = physics.get_state_indices()
+
+	Uq[:, :, irho] = rho
+	Uq[:, :, irhou] = rho * u
+	Uq[:, :, irhov] = rho * v
+	Uq[:, :, irhoE] = rhoE
+
+	# Calculate viscosity and thermal conductivity
+	mu, kappa = physics.get_transport(physics, Uq)
+	nu = mu / rho
+
+	# Get temperature
+	T = physics.compute_variable("Temperature", Uq, 
+		flag_non_physical=True)[0, 0, 0]
+
+	np.random.seed(10)
+	drdx = np.random.rand()
+	drdy = np.random.rand()
+	dudx = np.random.rand()
+	dudy = np.random.rand()
+	dvdx = np.random.rand() 
+	dvdy = np.random.rand()
+	divu = dudx + dvdy
+	dTdx = np.random.rand()
+	dTdy = np.random.rand()
+
+	tauxx = mu * (dudx + dudx - 2. / 3. * divu)
+	tauxy = mu * (dudy + dvdx)
+	tauyy = mu * (dvdy + dvdy - 2. / 3. * divu)
+
+	gUq = np.zeros([1, 1, ns, mesh.ndims])
+
+	gUq[:, :, irho, 0] = drdx
+	gUq[:, :, irhou, 0] = drdx * u + rho * dudx
+	gUq[:, :, irhov, 0] = drdx * v + rho * dvdx
+	gUq[:, :, irhoE, 0] = (R * T / (gamma - 1.) + \
+		0.5 * (u * u + v * v)) * drdx + rho * u * dudx \
+		+ rho * v * dvdx + rho * R / (gamma - 1.) * dTdx
+
+	gUq[:, :, irho, 1] = drdy
+	gUq[:, :, irhou, 1] = drdy * u + rho * dudy
+	gUq[:, :, irhov, 1] = drdy * v + rho * dvdy
+	gUq[:, :, irhoE, 1] = (R * T / (gamma - 1.) + \
+		0.5 * (u * u + v * v)) * drdy + rho * u * dudy \
+		+ rho * v * dvdy + rho * R / (gamma - 1.) * dTdy
+
+	Fref = np.zeros([1, 1, ns, 2])
+	Fref[:, :, irhou, 0] = tauxx
+	Fref[:, :, irhov, 0] = tauxy
+	Fref[:, :, irhoE, 0] = tauxx * u + tauxy * v + kappa * dTdx
+
+	Fref[:, :, irhou, 1] = tauxy
+	Fref[:, :, irhov, 1] = tauyy
+	Fref[:, :, irhoE, 1] = tauxy * u + tauyy * v + kappa * dTdy
```

Comment:
Try to think of more edge-case tests, as opposed to this, since this basically looks like a re-implementation of the actual flux code. Think of cases like, what if there are only gradients in one direction? Then set the gradient to an actual number (using random numbers is tricky because it's harder to hand calculate). Think of what this flux would be for small hand-calculatable problems like couette flow or something (incompressible so drdx and drdy are both zero, only dudy is nonzero, therefore tauxx must be zero, etc etc).

Reply:
I think adding a few of these tests in the future is a good idea, but I still think this one is good as it doesn't do the product rules for tauxx, tauxy, etc... It just calculates the equations as they are written and verifies that it matches our implementation. We also have the zero velocity cases as well to assist with this function.